### PR TITLE
Adjust selinux for proxmox

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec [hunk 1]
```diff
@@ -103,7 +103,16 @@ cp -pa .%{gem_dir}/* \
 %{foreman_restart}
 exit 0
 
+%post
+semanage port -a -t http_port_t -p tcp 8006
```
Comment (dgoetz) [inline packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: Should be `semanage port -a -t http_port_t -p tcp 8006 &> /dev/null || :` to suppress output and always run without error even if SELinux is disabled.

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec [hunk 2]
```diff
@@ -103,7 +103,16 @@ cp -pa .%{gem_dir}/* \
...
+
+%postun
+semanage port -d -t http_port_t -p tcp 8006
```
Comment (dgoetz) [inline packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: Should be `semanage port -d -t http_port_t -p tcp 8006 &> /dev/null || :` to suppress output and always run without error even if SELinux is disabled.

Comment (mdellweg) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: @dgoetz Thank you, changed.

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec [hunk 3]
```diff
@@ -103,7 +103,16 @@ cp -pa .%{gem_dir}/* \
 %{foreman_restart}
 exit 0
 
+%post
+semanage port -a -t http_port_t -p tcp 8006 &> /dev/null || :
+
+%postun
+semanage port -d -t http_port_t -p tcp 8006 &> /dev/null || :
```
Comment (ekohl) [inline packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: I thought postun was also called on upgrades. This means it'll flip flop between on/off/on. https://www.golinuxhub.com/2018/05/how-to-execute-script-at-pre-post-preun-postun-spec-file-rpm.html has some examples on how to check for this.

Comment (mdellweg) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: I guess, you are the expert on rpm packaging questions. `postun` sounded right to me, so i had no reason to question that. Feel free to add any needed change to this PR.

Comment (ekohl) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: I think technically an upgrade is an uninstall + install so it's called in both removals + upgrades. I also wouldn't consider myself an expert ;)

**Comment by dgoetz** [comment]
Give me some minutes and I will do the adjustments. I have working code in the icinga2 packages.

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec (hunk 3 above)
Comment (mdellweg) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: maybe prepending `[ $1 == 0 ] && ` is enough.

**Comment by dgoetz** [comment]
I added Requires section based on version as packages changed with Fedora 27 and RHEL 8 and the if condition which does the removal of the label only on uninstall of the last package (read real uninstall, but not upgrade).

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec (hunk 3 above)
Comment (mdellweg) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: When i read the article correctly, it is even install + uninstall. So my version would have left the port closed on update.
Thank you both for spotting and repairing this.

File: packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec [hunk 4]
```diff
@@ -34,6 +34,13 @@ BuildRequires: %{?scl_prefix_ruby}rubygems-devel
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 Provides: foreman-plugin-%{plugin_name} = %{version}
+%if 0%{?fedora} >= 27 || 0%{?rhel} >= 8
```
Comment (ekohl) [inline packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: Can you move it below the comment? Within the block it's autogenerated and replaced on package bumping so it'd be dropped on the next package update.

Comment (mdellweg) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: like this?

Comment (ekohl) [reply packages/plugins/rubygem-foreman_fog_proxmox/rubygem-foreman_fog_proxmox.spec]: Exactly.

**Comment by lzap** [comment]
> I wonder if this should be done in https://github.com/theforeman/foreman-selinux instead. @lzap?

I don't mind it here. Looks good.

**Pull request merged**